tests/garage/tf/models/test_model.py

```diff
@@ -304,3 +312,14 @@ def test_simple_model_is_pickleable_with_missing_parameters(self):
                 'SimpleModel/state/hidden_0/kernel:0']
             with pytest.warns(UserWarning):
                 model_pickled.build(state)
+
+    def test_model_set_parameters(self):
+        model1 = SimpleModel(output_dim=2, name='model1')
```

Comment:
hmm how does this work if the model is not in the top-level scope? i.e. what if it's in `/foo/bar/model1`?

Reply:
if model1 has `/foo/bar/model1/param1` and model2 has `/foo/bar/model2/param1`, then it match the parameters only looking at `/param1`. I think it's intuitive because the parameter name will be the sub-path of the model.